frontend/public/components/utils/firehose.jsx

```diff
@@ -163,7 +163,7 @@ export const Firehose = connect(
     }
 
     shouldComponentUpdate(nextProps, nextState, nextContext) {
-      const currentResources = this.props.resources;
+      const {resources: currentResources, forceUpdate = false} = this.props;
```

Comment:
Is there any other prop we can check for changes to avoid needing `forceUpdate`?

Reply:
The problem is that Firehose will not re-render if a prop changes on a child. We could define the props on Firehose itself since they get applied to it's children, but we would still need some way of checking arbitrary props in the `componentShouldUpdate` function. Though, that would break our definite types since we shouldn't just allow arbitrary props that aren't defined in the propTypes.

Reply:
Should I change Firehose to use `componentDidUpdate` instead of `shouldComponentUpdate`?

In the React docs for `componentDidUpdate`:

> This is also a good place to do network requests as long as you compare the current props to previous props

Reply:
Since currently, `shouldComponentUpdate` is essentially blocking re-rendering except when a few props change, I think it would be beneficial to use `componentDidUpdate` to just check for changes in the `resources` props to reset websockets.

Reply:
Let's open a tech debt issue and address it separately. Firehose is used so many places, I'd like to make sure we're careful with changes and don't want to block this PR.